jobrunner/cli/kill_job.py

```diff
@@ -55,20 +55,26 @@ def get_jobs(partial_job_ids):
     jobs = []
     need_confirmation = False
     for partial_job_id in partial_job_ids:
-        matches = database.find_where(Job, id__like=f"%{partial_job_id}%")
-        if len(matches) == 0:
-            raise RuntimeError(f"No jobs found matching '{partial_job_id}'")
-        elif len(matches) > 1:
-            print(f"Multiple jobs found matching '{partial_job_id}':")
-            for i, job in enumerate(matches, start=1):
-                print(f"  {i}: {job.slug}")
-            print()
-            index = int(input("Enter number: "))
-            assert 0 < index <= len(matches)
-            jobs.append(matches[index - 1])
+        # look for full matches
+        full_matches = database.find_where(Job, id=partial_job_id)
+        if len(full_matches) == 1:
+            jobs.append(full_matches[0])
         else:
-            need_confirmation = True
-            jobs.append(matches[0])
+            # look for partial matches
+            partial_matches = database.find_where(Job, id__like=f"%{partial_job_id}%")
+            if len(partial_matches) == 0:
+                raise RuntimeError(f"No jobs found matching '{partial_job_id}'")
+            elif len(partial_matches) > 1:
+                print(f"Multiple jobs found matching '{partial_job_id}':")
+                for i, job in enumerate(partial_matches, start=1):
+                    print(f"  {i}: {job.slug}")
+                print()
+                index = int(input("Enter number: "))
+                assert 0 < index <= len(partial_matches)
+                jobs.append(partial_matches[index - 1])
+            else:
+                need_confirmation = True
+                jobs.append(partial_matches[0])
```

Comment:
There's a simpler implementation here which doesn't involve adding a separate database query. You can just add a conditional around the line which sets the `need_confirmation` flag to only do so if there isn't an exact match e.g.
```py
# We only need confirmation if the supplied job ID doesn't exactly match the found job
if job.id != partial_job_id:
    need_confirmation = True
jobs.append(partial_matches[0])
```

In practical terms there's not much difference (the performance difference is basically irrelevant here), but it reduces the depth of nesting and I think is therefore a bit easier to understand.

Reply:
ah good idea, `job` isn't defined yet at that point, but this would work:
```
# We only need confirmation if the supplied job ID doesn't exactly match the found job
job = partial_matches[0]
if job.id != partial_job_id:
    need_confirmation = True
jobs.append(job)
```